core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,376 @@
+  var thumbnailCropper = element(
+    by.css('.protractor-test-photo-crop .cropper-container'));
+  var thumbnailUploadError = element(
+    by.css('.protractor-test-upload-error'));
+  var saveBlogPostContentButton = element(
+    by.css('.protractor-test-save-blog-post-content'));
+  var blogPostContentDisplay = element(
+    by.css('.protractor-test-content-display'));
+  var confirmButton = element(
+    by.css('.protractor-test-confirm-button'));
+  var currUsername = element(
+    by.css('.protractor-test-username-visible'));
+  var cancelThumbnailUploadButton = element(
+    by.css('.protractor-test-photo-upload-cancel'));
+  var blogDashboardIntroMessageContainer = element(
+    by.css('.protractor-test-intro-message-container'));
+  var closeBlogCardPreviewButton = element(
+    by.css('.protractor-test-close-preview-button'));
+  var blogCardPreviewButton = element(
+    by.css('.protractor-test-blog-card-preview-button'));
+  var blogDashboardLink = element(by.css(
+    '.protractor-test-blog-dashboard-link'));
+  var blogPostTiles = element.all(by.css(
+    '.protractor-test-blog-dashboard-tile'));
+  var blogPostTileTitle = by.css(
+    '.protractor-test-blog-post-title');
+  var navigateToBlogDashboardButton = element(
+    by.css('.protractor-test-back-button'));
+
+  this.get = async function() {
+    await waitFor.pageToFullyLoad();
+    await general.openProfileDropdown();
+    await action.click(
+      'Blog dashboard link from dropdown', blogDashboardLink);
+    await waitFor.pageToFullyLoad();
+    await waitFor.urlRedirection('http://localhost:9001/blog-dashboard');
+  };
+
+  this.navigateToBlogDashboardPageWithBackButton = async function() {
+    await action.click(
+      'Navigate back to blog dashboard button', navigateToBlogDashboardButton);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.waitForDraftBlogPostsToLoad = async function() {
+    await waitFor.visibilityOf(
+      draftBlogPostsTable, 'Blog posts table taking too long to appear.');
+  };
+
+  this.showAndCloseBlogCardPreview = async function() {
+    await action.click('Preview Button', blogCardPreviewButton);
+    await waitFor.visibilityOf(
+      currUserProfilePhoto,
+      'Current user profile photo taking too long to display');
+    await waitFor.visibilityOf(
+      currUsername,
+      'Current user name taking too long to display');
+    await action.click('Close Preview', closeBlogCardPreviewButton);
+  };
+
+  this.waitForPublishedBlogPostsToLoad = async function() {
+    await waitFor.visibilityOf(
+      publishedBlogPostsTable, 'Blog posts table taking too long to appear.');
+  };
+
+  this.createNewBlogPost = async function() {
+    await action.click('Create Blog Post button', createBlogPostButton);
+
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(
+      blogPostTitleFieldElement,
+      'Blog Post Editor is taking too long to appear.');
+    await waitFor.pageToFullyLoad();
```

Comment:
Why are we waiting for the page to load here?

Reply:
So as to ensure that any other edit is done only after the editor page has been fully loaded.